weave-js/track-v1.js

```diff
@@ -0,0 +1,185 @@
+      ];
+      analytics.factory = function (t) {
+        return function () {
+          var e = Array.prototype.slice.call(arguments);
+          e.unshift(t);
+          analytics.push(e);
+          return analytics;
+        };
+      };
+      for (var t = 0; t < analytics.methods.length; t++) {
+        var e = analytics.methods[t];
+        analytics[e] = analytics.factory(e);
+      }
+    }
+
+    if (window.document.cookie.indexOf('segment_logging') != -1) {
+      enableSegmentAnalyticsLogging();
+    }
+  }
+
+  addSegmentAnalyticsMethods();
+
+  function enableSegmentAnalyticsLogging() {
+    const analytics = window.analytics;
+    Object.getOwnPropertyNames(analytics)
+      .filter(p => {
+        return typeof analytics[p] === 'function';
+      })
+      .forEach(p => {
+        const f = analytics[p];
+        analytics[p] = function () {
+          if (p === 'page') {
+            console.groupCollapsed('segment:' + p + ' ' + arguments[1]);
+            console.log('category: ', arguments[0]);
+            console.log('properties: ', arguments[2]);
+            console.groupEnd();
+          } else if (p === 'track' || p === 'group' || p === 'identify') {
+            console.groupCollapsed('segment:' + p + ' ' + arguments[0]);
+            console.log('properties: ', arguments[1]);
+            console.groupEnd();
+          }
+          return f.apply(this, arguments);
+        };
+      });
+  }
+
+  function analyticsCallback() {
+    if (window.document.cookie.indexOf('segment_logging') != -1) {
+      enableSegmentAnalyticsLogging();
+    }
+  }
+
+  function enableSegmentAnalytics() {
+    // Adds the segment analytics library. This is boilerplate.
+    if (!analytics.initialize) {
+      analytics.SNIPPET_VERSION = '4.0.0';
+      // this is segment's analytics.js library (https://github.com/segmentio/analytics.js-core) bundled with all of our integrations and settings.
+      // it can be retrieved from https://cdn.segment.com/analytics.js/v1/WmJvIs1MTqNjKAeQmEyw6TvqyRI5Su2z/analytics.min.js
+      // we serve it locally since cdn.segment.com is usually blocked by ad-blockers.
+      insertCookieControlledScript('/sa.min.js', 2, () => analyticsCallback());
+
+      // autotrack.js provides the trackers required below
+      insertCookieControlledScript('/autotrack.js', 2);
+      window.ga =
+        window.ga ||
+        function () {
+          window.ga.q = window.ga.q || [];
+          window.ga.q.push(arguments);
+        };
+      window.ga('require', 'eventTracker');
+      window.ga('require', 'outboundLinkTracker');
+      window.ga('require', 'urlChangeTracker');
+      window.ga('require', 'cleanUrlTracker');
+      window.ga('require', 'impressionTracker');
+      window.ga('require', 'maxScrollTracker');
+      window.ga('require', 'mediaQueryTracker');
+      window.ga('require', 'outboundFormTracker');
+      window.ga('require', 'pageVisibilityTracker');
+    }
+  }
+
+  function enablePendo() {
```

Comment:
do we want/need pendo?

Reply:
We can always turn it off so I'm of the opinion that we should turn it on -- they might provide some interesting graphs/metrics out of the box. It can also serve as a double check on our page view data numbers.